[ML] Extend reindexed AD results template and generalize heal beyond job_id (#153755)
elastic/elasticsearch: docs/reference/elasticsearch/configuration-reference/machine-learning-settings.md

:   ([Dynamic](docs-content://deploy-manage/stack-settings.md#dynamic-cluster-setting)) The maximum size the anomaly detection results indices can reach before being rolled over by the nightly maintenance task. When the {{operator-feature}} is enabled, this setting can be updated only by operator users. Valid values must be greater than or equal to `-1B`. A value of `-1B` means the indices will never be rolled over. A value of `0B` means the indices will always be rolled over, regardless of size. Defaults to `50GB`.

`xpack.ml.anomalies.heal_reindexed_v7.enabled` {applies_to}`stack: ga 9.3`

> :   ([Dynamic](docs-content://deploy-manage/stack-settings.md#dynamic-cluster-setting)) Controls if {{es}} automatically fixes anomaly detection result aliases that point to indices matching `.reindexed-*-ml-anomalies-*` with incorrect mappings after an upgrade. See [#147686](https://github.com/elastic/elasticsearch/issues/147686) for details. When `true`, startup maintenance repairs affected aliases; historic data remains in old indices, with ML Notifications providing manual reindexing steps. Set to `false` to disable. Defaults to `true`.

Reviewer: ```suggestion
:   ([Dynamic](docs-content://deploy-manage/stack-settings.md#dynamic-cluster-setting)) Controls whether {{es}} automatically fixes {{anomaly-detect}} result aliases that point to indices matching `.reindexed-v7-ml-anomalies-*` with incorrect mappings after an upgrade. For details, refer to [#147686](https://github.com/elastic/elasticsearch/issues/147686).
```

Author: Applied the Vale/style fixes (whether, anomaly-detect, turn off). Kept the `.reindexed-*-ml-anomalies-*` wildcard reference rather than v7-only because the heal now covers v7 and v8 lineages and fields beyond `job_id` — kosabogi suggestion was written before that generalization.
